packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -1881,33 +1883,47 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
   Future<T> pushAndRemoveUntil<T extends Object>(Route<T> newRoute, RoutePredicate predicate) {
     assert(!_debugLocked);
     assert(() { _debugLocked = true; return true; }());
-    final List<Route<dynamic>> removedRoutes = <Route<dynamic>>[];
-    while (_history.isNotEmpty && !predicate(_history.last)) {
-      final Route<dynamic> removedRoute = _history.removeLast();
```

Comment:
Is there any reason to no longer collect the to-be-removed routes before-hand? Seems like that would simplify matters a little bit.

Reply:
Fixed.